File: src/IO/dislin-11.0/python/contour.py
#! /usr/bin/env python
import math
import dislin

ctit1 = 'Contour Plot'
ctit2 = 'F(X,Y) = 2 * SIN(X) * SIN (Y)'

n = 50
m = 50

xray = range (n)
yray = range (m)
zmat = range (n*m)

fpi  = 3.1415927 / 180.
stepx = 360. / (n - 1)
stepy = 360. / (m - 1)

for i in range (0, n):
  xray[i] = xray[i] * stepx

for i in range (0, m):
  yray[i] = yray[i] * stepy

for i in range (0, n):
  x = xray[i] * fpi
  for j in range (0, m):
    y = yray[j] * fpi
    zmat[i*m+j] = 2 * math.sin(x) * math.sin(y)

dislin.metafl ('cons')
dislin.setpag ('da4p')
dislin.disini ()
dislin.pagera ()
dislin.complx ()

dislin.titlin (ctit1, 1)
dislin.titlin (ctit2, 3)

dislin.intax  ()
dislin.axspos (450, 2650)

dislin.name   ('X-axis', 'X')
dislin.name   ('Y-axis', 'Y')

dislin.graf   (0., 360., 0., 90., 0., 360., 0., 90.)
dislin.height (50)
dislin.title  ()

dislin.height (30)
for i in range (0, 9):
  zlev = -2. + i * 0.5
  if i == 4:
    dislin.labels ('NONE', 'CONTUR')
  else:
    dislin.labels ('FLOAT', 'CONTUR')

  dislin.setclr ((i+1) * 28)
  dislin.contur (xray, n, yray, m, zmat, zlev)

dislin.disfin ()

File: src/IO/dislin-11.0/java/contour.java
import de.dislin.Dislin;

public class contour {
  public static void main (String args []) {
     int n = 50, m = 50, i, j;
     float zlev;
     float zmat []  = new float [n*m];
     float xray []  = new float [n];
     float yray []  = new float [m];

     String ctit1 = "Contour Plot";
     String ctit2 = "F(X,Y) = 2*SIN(X)*SIN(Y)";

     double x, y;
     double fpi = 3.1415926/180.;
     double stepx = 360. / (n-1);
     double stepy = 360. / (m-1);

     for (i = 0; i < n; i++) {
        xray[i] = (float) (i * stepx);
     }

     for (j = 0; j < m; j++) {
        yray[j] = (float) (j * stepy);
     }

     for (i = 0; i < n; i++) {
       x = xray[i] * fpi;
       for (j = 0; j < m; j++) {
         y = yray[j] * fpi;
         zmat[i*m+j] = (float) (2 * Math.sin(x)* Math.sin(y));
       }
     }
       
     Dislin.metafl ("cons");
     Dislin.setpag ("da4p");
     Dislin.disini ();
     Dislin.pagera ();
     Dislin.complx ();

     Dislin.titlin (ctit1, 1);
     Dislin.titlin (ctit2, 3);
     Dislin.intax  ();
     Dislin.axspos (450, 2650);
         
     Dislin.name   ("X-axis", "x");
     Dislin.name   ("Y-axis", "y");

     Dislin.graf   (0.f, 360.f, 0.f, 90.f,
                    0.f, 360.f, 0.f, 90.f);
     Dislin.height (50);
     Dislin.title  ();

     Dislin.height (30);
     for (i = 0; i < 9; i++) {
       zlev = -2.f + i * 0.5f;
       if (i == 4) {
         Dislin.labels ("none", "contur");
       }
       else {
         Dislin.labels ("float", "contur");
       }
       Dislin.setclr ((i+1) * 28);
       Dislin.contur (xray, n, yray, m, zmat, zlev);
     }
     Dislin.disfin ();
  }
}

